demos/browser/app/meeting/meeting.ts

```diff
@@ -475,27 +475,51 @@ export class DemoMeetingApp implements AudioVideoObserver, DeviceChangeObserver
     });
   }
 
-  getNearestMediaRegion(): void {
-    const supportedMediaRegions: Array<string> = ['ap-northeast-1', 'ap-southeast-1', 'ap-southeast-2', 'ca-central-1', 'eu-central-1', 'eu-north-1', 'eu-west-1', 'eu-west-2', 'eu-west-3', 'sa-east-1', 'us-east-1', 'us-east-2', 'us-west-1', 'us-west-2'];
+
+  getSupportedMediaRegions(): Array<string> {
+    const supportedMediaRegions: Array<string> = [];
+    const mediaRegion = (document.getElementById("inputRegion")) as HTMLSelectElement;
+    for (var i = 0; i < mediaRegion.length; i++) {
+      supportedMediaRegions.push(mediaRegion.value);
+    }
+    return supportedMediaRegions;
+  }
+
+  async getNearetMediaRegion(): Promise<string> {
+    var nearestMediaRegion = '';
+    const defaultMediaRegion = 'us-east-1';
+    try {
+      const nearestMediaRegionResponse = await fetch(
+        `https://nearest-media-region.l.chime.aws`,
+        {
+          method: 'GET',
+        }
+      );
+      const nearestMediaRegionJSON = await nearestMediaRegionResponse.json();
+      this.log(nearestMediaRegionJSON.region);
+      nearestMediaRegion = nearestMediaRegionJSON.region;
+    } catch (error) {
+      nearestMediaRegion = defaultMediaRegion;
+      this.log('Default media region ' + defaultMediaRegion + ' selected: ' + error.message);
```

Comment:
Let's remove this. We dont want any default logic in this.
Also let's move all the logging as well as try and catch error to setMediaRegion. This function should just return nearestMediaRegionResponse.json().region